components/ai_chat/core/browser/engine/conversation_api_client_unittest.cc

```diff
@@ -0,0 +1,346 @@
+               const std::string&,
+               const std::string&,
+               DataReceivedCallback,
+               ResultCallback,
+               (const base::flat_map<std::string, std::string>&),
+               const api_request_helper::APIRequestOptions&),
+              (override));
+};
+
+// Create a version of the ConversationAPIClient that contains our mocks
+class TestConversationAPIClient : public ConversationAPIClient {
+ public:
+  explicit TestConversationAPIClient(
+      AIChatCredentialManager* credential_manager)
+      : ConversationAPIClient("unit_test_model_name",
+                              nullptr,
+                              credential_manager) {
+    SetAPIRequestHelperForTesting(std::make_unique<MockAPIRequestHelper>(
+        net::NetworkTrafficAnnotationTag(TRAFFIC_ANNOTATION_FOR_TESTS),
+        nullptr));
+  }
+  ~TestConversationAPIClient() override = default;
+
+  MockAPIRequestHelper* GetMockAPIRequestHelper() {
+    return static_cast<MockAPIRequestHelper*>(GetAPIRequestHelperForTesting());
+  }
+};
+
+class ConversationAPIUnitTest : public testing::Test {
+ public:
+  ConversationAPIUnitTest() = default;
+  ~ConversationAPIUnitTest() override = default;
+
+  void SetUp() override {
+    credential_manager_ = std::make_unique<MockAIChatCredentialManager>(
+        base::NullCallback(), &prefs_service_);
+    client_ =
+        std::make_unique<TestConversationAPIClient>(credential_manager_.get());
+    // Intercept credential fetch
+    EXPECT_CALL(*credential_manager_, FetchPremiumCredential(_))
+        .WillOnce(
+            [&](base::OnceCallback<void(std::optional<CredentialCacheEntry>)>
+                    callback) {
+              std::move(callback).Run(std::move(credential_));
+            });
+  }
+
+  void TearDown() override {}
+
+  std::string GetEventsJson(std::string_view body_json) {
+    auto dict = base::JSONReader::ReadDict(body_json);
+    EXPECT_TRUE(dict.has_value());
+    base::Value::List* events = dict->FindList("events");
+    EXPECT_TRUE(events);
+    std::string events_json;
+    base::JSONWriter::WriteWithOptions(
+        *events, base::JSONWriter::OPTIONS_PRETTY_PRINT, &events_json);
+    return events_json;
+  }
+
+  std::string FormatComparableEventsJson(std::string_view formatted_json) {
+    auto events = base::JSONReader::Read(formatted_json);
+    EXPECT_TRUE(events.has_value()) << "Verify that the string is valid JSON!";
+    std::string events_json;
+    base::JSONWriter::WriteWithOptions(
+        events.value(), base::JSONWriter::OPTIONS_PRETTY_PRINT, &events_json);
+    return events_json;
+  }
+
+ protected:
+  base::test::TaskEnvironment task_environment_;
+  std::unique_ptr<TestConversationAPIClient> client_;
+  std::unique_ptr<MockAIChatCredentialManager> credential_manager_;
+  TestingPrefServiceSimple prefs_service_;
+  std::optional<CredentialCacheEntry> credential_ = std::nullopt;
+};
+
+TEST_F(ConversationAPIUnitTest, PremiumHeaders) {
```

Comment:
PerormRequest_PremiumHeaders and add the same prefix for other test cases.

Reply:
Modified the test names (and now squashed all follow-ups now). Thanks for the review!